### PR TITLE
[helm chart] Update to use cluster-autoscaler 1.26.2 + default tag value from the Chart's AppVersion

### PR DESCRIPTION
**Pull request by asherf**
This is similar to how it is done in other chats:
https://github.com/kubernetes/dashboard/blob/313af7c8e1922fb4d7c6be4dc3fc006e286e759a/charts/helm-chart/kubernetes-dashboard/templates/deployment.yaml#L66 https://github.com/kubernetes/dashboard/blob/313af7c8e1922fb4d7c6be4dc3fc006e286e759a/charts/helm-chart/kubernetes-dashboard/values.yaml#L23

/kind feature

### CODE REVIEW

**Comment by asherf** [comment]
not sure about the docs lint fail.... any pointers ?

File: charts/cluster-autoscaler/values.yaml [hunk 1]
```diff
@@ -229,8 +229,7 @@ hostNetwork: false
 image:
   # image.repository -- Image repository
   repository: registry.k8s.io/autoscaling/cluster-autoscaler
-  # image.tag -- Image tag
-  tag: v1.24.0
+  tag: ""  # If not defined, uses appVersion of Chart.yaml
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/values.yaml:232]:
```suggestion
  # image.tag - Image tag - defaults to the version defined in Chart.AppVersion
  tag: ""  # If not defined, uses appVersion of Chart.yaml
```

**Review by gjtempleton — CHANGES_REQUESTED** [review]
Hey, thanks for the PR.

We use and enforce helm-docs to generate our chart README, which is why the CI is currently failing, you can either set it up as a [pre-commit hook](https://github.com/kubernetes/autoscaler/tree/master/charts#pre-commit-hooks) or run it manually.

**Comment by gjtempleton** [comment]
On balance I'm not sure this is a path we want to take with the CA given versions of the CA are so closely tied to the version of kubernetes and mismatching minor versions between these can result in unexpected behaviour.

There's been a previous request to look at automatically picking an image to match the minor version of the underlying cluster as per #5324.

**Comment by asherf** [comment]
> On balance I'm not sure this is a path we want to take with the CA given versions of the CA are so closely tied to the version of kubernetes and mismatching minor versions between these can result in unexpected behaviour.
> 
> There's been a previous request to look at automatically picking an image to match the minor version of the underlying cluster as per #5324.

good point.
closing in favor of https://github.com/kubernetes/autoscaler/pull/5640#issuecomment-1493723676
I appreciate the feedback and the guidance on using pre-commit, I somehow missed that.

**Pull request closed without merging**